posthog/warehouse/models/table.py

```diff
@@ -0,0 +1,104 @@
+from posthog.models.utils import UUIDModel, CreatedMetaFields, sane_repr, DeletedMetaFields
+from posthog.errors import wrap_query_error
+from django.db import models
+from posthog.models.team import Team
+from posthog.client import sync_execute
+from .credential import DataWarehouseCredential
+from posthog.hogql.database.models import (
+    StringDatabaseField,
+    IntegerDatabaseField,
+    DateTimeDatabaseField,
+    StringJSONDatabaseField,
+)
+from posthog.hogql.database.s3_table import S3Table
+
+ClickhouseHogqlMapping = {
+    "String": StringDatabaseField,
+    "DateTime64": DateTimeDatabaseField,
+    "DateTime32": DateTimeDatabaseField,
+    "UInt8": IntegerDatabaseField,
+    "UInt16": IntegerDatabaseField,
+    "UInt32": IntegerDatabaseField,
+    "UInt64": IntegerDatabaseField,
+    "Float8": IntegerDatabaseField,
+    "Float16": IntegerDatabaseField,
+    "Float32": IntegerDatabaseField,
+    "Float64": IntegerDatabaseField,
+    "Int32": IntegerDatabaseField,
+    "Int64": IntegerDatabaseField,
+    "Tuple": StringJSONDatabaseField,
+    "Array": StringJSONDatabaseField,
+    "Map": StringJSONDatabaseField,
```

Comment:
These are actually not correct. The `StringJSONDatabaseField` expects a literal string with JSON inside, like our `properties` fields. However there's no better type I can give you now. These types will probably just not work for the moment 🤷